### PR TITLE
Save state in new format

### PR DESCRIPTION
**Pull request by bitdivision**
### What is the context of this PR?
DO NOT MERGE until https://github.com/ONSdigital/eq-survey-runner/pull/1921 is in production!

Now that https://github.com/ONSdigital/eq-survey-runner/pull/1905 has gone into production, we need to enable writing of new formats.

This PR essentially reverts the changes from https://github.com/ONSdigital/eq-survey-runner/pull/1905. It adds this as version 3 of the questionnaire store.

- Data is written compressed / without base64 and without a JSON wrapper.

### How to review 

Test that nothing goes wrong when:
- starting a questionnaire on version 2 (with #1905 and #1921 )
- upgrade to version 3
- continue the questionnaire
- downgrade to version 2
- continue the questionnaire

At each stage of the above, we should probably test submission.

### Checklist

* [ ] New static content marked up for translation
* [ ] Newly defined schema content included in eq-translations repo

### CODE REVIEW

**Review by ajmaddaford — APPROVED** [review]

File: app/views/questionnaire.py [hunk 1]
```diff
@@ -436,7 +436,7 @@ def submit_answers(routing_path, eq_id, form_type, schema):
     _store_submitted_time_in_session(submitted_time)
 
     if is_view_submitted_response_enabled(schema.json):
-        _store_viewable_submission(list(answer_store), metadata, submitted_time)
+        _store_viewable_submission(answer_store.answer_map, metadata, submitted_time)
```
Comment (ajmaddaford) [inline app/views/questionnaire.py]: This should probably be in the another pull request as it's not related to saving the new compressed questionnaire state.

**Review by ajmaddaford — CHANGES_REQUESTED** [review]

**Review by ajmaddaford — APPROVED** [review]

**Review by MebinAbraham — APPROVED** [review]

File: tests/integration/session/test_questionnaire_store_upgrade.py [hunk 2]
```diff
@@ -28,15 +28,15 @@ class TestLogin(IntegrationTestCase):
 
     def test_questionnaire_store_is_upgraded(self):
         # Given
-        previous_version = QuestionnaireStore.LATEST_VERSION - 1
 
-        # Creates a QuestionnaireStore with previous version
-        with patch('app.data_model.questionnaire_store.QuestionnaireStore.get_latest_version_number', return_value=previous_version):
-            self.launchSurvey('test', '0205')
+        self.launchSurvey('test', '0205')
```
Comment (dwyeradam) [inline tests/integration/session/test_questionnaire_store_upgrade.py]: Does this need to use test_0205.json? Ideally we should be getting rid of these and using actual test schemas. Not a blocker as it's not related to this PR.

Comment (bitdivision) [reply tests/integration/session/test_questionnaire_store_upgrade.py]: Fixed now.
